src/OpenLoco/src/Vehicles/Vehicle.cpp

```diff
@@ -826,5 +826,90 @@ namespace OpenLoco::Vehicles
                 regs.esi = X86Pointer(newComponent);
                 return 0;
             });
+
+        registerHook(
+            0x004AF5E1,
+            [](registers& regs) FORCE_ALIGN_ARG_POINTER -> uint8_t {
+                registers backup = regs;
+                VehicleHead* head = X86Pointer<VehicleHead>(regs.esi);
+                connectJacobsBogies(*head);
+                regs = backup;
+                return 0;
+            });
+    }
+
+    // remember to move this above registerHooks after rebasing
+    // 0x004AF5E1
+    // esi: head
+    // returns nothing
+    void connectJacobsBogies(VehicleHead& head)
+    {
+        // if jacob's bogie is available at front the front body will have jacobsBogieAvailable
+        // if jacob's bogie is available at back the end body will have jacobsBogieAvailable
+        Vehicle train(head);
+        auto componentsFound = 0;
+        CarComponent previousCarComponent;
+        CarComponent secondPreviousCarComponent;
+        for (auto& car : train.cars)
```

Comment:
what happens with two jacobs bogies cars in one train? My reading of this is that jacobs bogies can only be between a cars carComponents.

Reply:
To create a jacob's bogie connection, two sequential CarComponent's bodies need the flag set. This can only occur if two Cars with jacob's bogie connections are sequential, because of a number of factors the code to create jacob's bogies only fires on the first CarComponent of a car (by checking body_start).